src/protocol/context.rs

```diff
@@ -27,7 +28,7 @@ pub struct ProtocolContext<'a, N, F> {
     accumulator: Option<SecurityValidatorAccumulator<F>>,
 }
 
-impl<'a, N, F> ProtocolContext<'a, N, F> {
+impl<'a, N, F: Field> ProtocolContext<'a, N, F> {
```

Comment:
I should have commented on #164, but should `ProtocolContext` be generic over `F`? We have one protocol, `ConvertShares`, which takes `XorShares` as its input. It isn't causing any issue now because the context in ConvertShares is only used to identify the helper, but using a context that takes an `F` to execute on shares that are not fields feels odd. It also forces `make_contexts` to return contexts with `Fp31`, but the code that generates the contexts is agnostic about this.

Would it be possible to pass `SecurityValidatorAccumulator` and specify an `F` when calling `MaliciouslySecureMul::execute()`?

Reply:
I created #180, which removes `F` from `ProtocolContext`, just to illustrate as an example. If we actually need `F`, please disregard. Otherwise, I'll be happy to rebase on top of this once merged, make more edits if necessary, and push again.

Reply:
As discussed on #180, I think we will need <F> for now, since we will not have access to the `SecurityValidatorAccumulator` within all the protocols which invoke multiplications. As such, it's easiest to just get it from the ProtocolContext. And the ProtocolContext will be validating entire protocols over a single field (one that is used for information-theoretic MACs).